Лабораторная работа 2

vitalik1609 opened the pull request:
<img width="347" alt="Снимок экрана 2024-10-11 в 22 11 17" src="https://github.com/user-attachments/assets/ff761d4c-3db5-495c-affd-98aca2b745f3">

vitalik1609 committed:
Add files via upload

dmitrylaabe commented on Лаб 2/task2.py:
```diff
@@ -0,0 +1,12 @@
+
+# TODO Рассчитайте подушку безопасности, чтобы протянуть 10 месяцев без долгов
+costs = 0
+for i in range(1, months+1):
```
Изящнее range(months) использовать

dmitrylaabe commented on Лаб 2/task2.py:
```diff
@@ -0,0 +1,12 @@
+# TODO Рассчитайте подушку безопасности, чтобы протянуть 10 месяцев без долгов
+costs = 0
+for i in range(1, months+1):
+    costs += abs(spend-salary)
```
А в чем алгоритмический смысл abs()? Ведь с точки зрения здравого смысла, если зарплата больше затрат, то подушка безопасности не нужна, а точнее даже наоборот- происходит накопление. В costs мы фактически считаем дефицит, который сформируется за 10 месяцев, а накопления с точки зрения дефицита имеют обратный знак, так что если решать задачу в общем случае, то abs не нужен, а costs может сначала уменьшаться и быть отрицательным, а потом нарастать. И правильно после цикла проверить, какой он получился - если отрицательный, то подушка безопасности не нужна, а если положительный, то выводим его, как результат.

vitalik1609 committed:
Update task2.py

Теперь в costs считаем остаток от зп после трат. Если по итогу отрицательный, то нужна подушка безопасности обратная по знаку. Если остаток не отрицательный то подушка не нужна
